src/usr/local/www/diag_backup.php

```diff
@@ -157,7 +157,7 @@ function check_and_returnif_section_exists($section) {
 	if ($mode) {
 		if ($mode == "download") {
 			if ($_POST['encrypt']) {
-				if (!$_POST['encrypt_password']) {
+				if (!$_POST['encrypt_password'] || ($_POST['encrypt_password'] != $_POST['encrypt_password_confirm'])) {
 					$input_errors[] = gettext("A password for encryption must be supplied and confirmed.");
```

Comment:
Suggest: "Supplied password and confirmation do not match."

Reply:
np, replaced